### PR TITLE
Clean up map.c /map.h

### PR DESCRIPTION
**Pull request by steffen-kiess**
These commits remove all manual changes from map.c / map.h (which should be autogenerated by create-native-map).

### CODE REVIEW

**Comment by marek-safar** [comment]
approve

**Comment by marek-safar** [comment]
@lambdageek please review

**Comment by steffen-kiess** [comment]
The windows build was broken because I didn't remove `Mono_Posix_FromMremapFlags` and `Mono_Posix_ToMremapFlags` from `msvc/monoposixhelper.def`, should be fixed now.

File: support/stdio.c [hunk 1]
```diff
@@ -202,7 +202,7 @@ Mono_Posix_Stdlib_fgetc (void* stream)
 	return fgetc (stream);
 }
 
-char*
+void*
```
Comment (lambdageek) [inline support/stdio.c]: Why `void*`?  `fgets` returns a `char*`

Comment (steffen-kiess) [reply support/stdio.c]: The return type has to match the prototype in `map.h`, and the prototype in `map.h` is autogenerated from `Stdlib.sys_fgets` which returns `IntPtr` (which is mapped to `void*`): https://github.com/mono/mono/blob/ad2132e611a7e8b7c7dcaed1c3ecc23d40dc17de/mcs/class/Mono.Posix/Mono.Unix.Native/Stdlib.cs#L888

Comment (akoeplinger) [reply support/stdio.c]: We only check the pointer against null in Stdlib.cs so the type isn't really relevant :)

**Comment by akoeplinger** [comment]
Thanks for the contribution!

**Pull request merged**